Food-Waste-Optimization/Food-Waste-Optimization
# Add docker container for server and client

**Files modified**

- src/index.py : Flask app should be accessible on all addresses inside of the container, so the host = 0.0.0.0 should be set.
- src/frontend/vite.config.js: Configured to work also in container. Frontend accessible on port 3000 while running in container.
- docker-compose.yml: Added services for both backend and frontend. Easily runnable now from docker.

**Added files**

- Dockerfile renamed to Dockerfile.backend: Describing image for flask webserver.
- Dockerfile.frontend: Describing image for frontend / React app


**How to test on your machine that it really works**

- Read carefully the docker-compose.yml file and set the appropriate path for data according to its location inside your machine.
- While being in the root directory of the project run following commands
    - `docker compose up`  which builds and runs the containers for both frontend and backend.
- Now make sure you can access backend from **localhost:5000** and frontend from **localhost:3000**
